### PR TITLE
feature(new integration): revamp custify

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change

Revamping Custify for extended support towards group call, reducing cognitive complexity and general improvements

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/custify/data/CUSTIFYIdentifyConfig.json [hunk 1]
```diff
@@ -1,80 +1,62 @@
...
+      "traits.signedUpAt",
+      "context.traits.signedUpAt",
+      "timestamp",
+      "originalTimestamp"
     ],
-    "required": false,
-    "metadata": {
-      "type": "secondTimestamp"
-    }
+    "required": false
```
Comment (ItsSudip) [inline v0/destinations/custify/data/CUSTIFYIdentifyConfig.json:55]: We have to add metadata to convert timestamp to Unix.

File: v0/destinations/custify/data/CUSTIFYTrackConfig.json [hunk 2]
```diff
@@ -1,30 +1,14 @@
...
-    ],
-    "required": false
-  },
-  {
-    "destKey": "company_id",
-    "sourceKeys": [
-      "groupId",
-      "traits.groupId",
-      "context.traits.groupId"
-    ],
```
Comment (ItsSudip) [inline v0/destinations/custify/data/CUSTIFYTrackConfig.json:19]: Don't we need to send `conpany_id` in track calls?

File: v0/destinations/custify/data/CUSTIFYTrackConfig.json [hunk 3]
```diff
@@ -35,12 +19,10 @@
   {
     "destKey": "created_at",
     "sourceKeys": [
-      "originalTimestamp",
-      "timestamp"
+      "timestamp",
+      "originalTimestamp"
+
```
Comment (ItsSudip) [inline v0/destinations/custify/data/CUSTIFYTrackConfig.json:24]:
```suggestion
```

File: v0/destinations/custify/deleteUsers.js [hunk 4]
```diff
@@ -1,44 +1,44 @@
...
 
 const userDeletionHandler = async (userAttributes, config) => {
+  if (!userAttributes) {
+    throw new CustomError("userAttributes for deletion not present", 400);
+  }
+  const { apiKey } = config;
+  const { userId } = userAttributes;
   if (!config) {
     throw new CustomError("Config for deletion not present", 400);
   }
```
Comment (ItsSudip) [inline v0/destinations/custify/deleteUsers.js:15]:
```suggestion
  if (!config) {
    throw new CustomError("Config for deletion not present", 400);
  }
  const { apiKey } = config;
  const { userId } = userAttributes;
  
```

File: v0/destinations/custify/util.js [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    throw new CustomError("Email or userId is mandatory", 400);
+  }
+  if (userPayload.name === undefined || userPayload.name === "") {
+    const firstName = getFieldValueFromMessage(message, "firstName");
+    const lastName = getFieldValueFromMessage(message, "lastName");
+    if (firstName && lastName) {
+      finalPayload.name = `${firstName} ${lastName}`;
+    } else {
+      finalPayload.name = firstName || lastName;
+    }
```
Comment (ItsSudip) [inline v0/destinations/custify/util.js:82]: What if both are missing?

File: v0/destinations/custify/deleteUsers.js (hunk 4 above)
Comment (utsabc) [reply v0/destinations/custify/deleteUsers.js:15]: good catch updating

File: v0/destinations/custify/data/CUSTIFYTrackConfig.json (hunk 2 above)
Comment (utsabc) [reply v0/destinations/custify/data/CUSTIFYTrackConfig.json:19]: Yeah we need along with these due merge it didnt came

File: v0/destinations/custify/data/CUSTIFYIdentifyConfig.json (hunk 1 above)
Comment (utsabc) [reply v0/destinations/custify/data/CUSTIFYIdentifyConfig.json:55]: We are not using secondsTrime stamp anymore

File: v0/destinations/custify/util.js (hunk 5 above)
Comment (utsabc) [reply v0/destinations/custify/util.js:82]: it will be set as undefined and get cleared at the end

**Review by ItsSudip — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Pull request merged**